plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/AmazonEC2Mock.java

```diff
@@ -1144,6 +1162,11 @@ public ModifyClientVpnEndpointResult modifyClientVpnEndpoint(ModifyClientVpnEndp
         throw new UnsupportedOperationException("Not supported in mock");
     }
 
+    @Override
+    public ModifyEbsDefaultKmsKeyIdResult modifyEbsDefaultKmsKeyId(ModifyEbsDefaultKmsKeyIdRequest modifyEbsDefaultKmsKeyIdRequest) {
```

Comment:
Had to add a few new overrides here since the API got expanded

Reply:
I think we should not add new overrides but instead extend `AbstractAmazonEC2` and clean up all unnecessary overridden methods.

Reply:
Wow 3k lines saved from that, thanks!

Reply:
Nice!